src/services/jsDoc.ts

```diff
@@ -169,7 +169,22 @@ namespace ts.JsDoc {
             case SyntaxKind.JSDocAugmentsTag:
                 return withNode((tag as JSDocAugmentsTag).class);
             case SyntaxKind.JSDocTemplateTag:
-                return addComment((tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", "));
+                const templateTag = tag as JSDocTemplateTag;
+                const displayParts: SymbolDisplayPart[] = [];
+                if (templateTag.constraint) {
+                    displayParts.push(textPart(templateTag.constraint.getText()));
+                }
+                if (length(templateTag.typeParameters)) {
+                    if (length(displayParts)) {
+                        displayParts.push(spacePart());
+                    }
+                    displayParts.push(namePart(map(templateTag.typeParameters, tp => tp.getText()).join(", ")));

```

Comment:
Won't this classify `T, U` as one `namePart`? Seems like you want to use `intersperse`

https://github.com/microsoft/TypeScript/blob/2ce05a82920aced69f31bb3ab9e7d3e710980e91/src/compiler/core.ts#L132-L146

which is already used in the same file already: https://github.com/microsoft/TypeScript/blob/aee779ac7d4eabba195e1097d05e54332e9e9d35/src/services/jsDoc.ts#L114

Reply:
The `typeParameters` conversion was not changed. It is the same as it was before.
```ts
(tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", ")
```
Should `typeParameters` be returned as an array?

Reply:
I think you want

```suggestion
                    for (let i = 0, n = templateTag.typeParameters.length; i < n; i++) {
                        const tp = templateTag.typeParameters[i];
                        displayParts.push(namePart(tp.getText());
                        if (i !== n - 1) {
                            displayParts.push(punctuationPart(SyntaxKind.CommaToken));
                            displayParts.push(spacePart());
                        }
                    }
```

It's possible you should use `typeParameterNamePart`

```suggestion
                    for (let i = 0, n = templateTag.typeParameters.length; i < n; i++) {
                        const tp = templateTag.typeParameters[i];
                        displayParts.push(typeParameterNamePart(tp.getText());
                        if (i !== n - 1) {
                            displayParts.push(punctuationPart(SyntaxKind.CommaToken));
                            displayParts.push(spacePart());
                        }
                    }
```

Reply:
(it's possible that I'm missing something though - feel free to correct me if so)

Reply:
`namePart` uses `getTagNameDisplayPart` which handles this
https://github.com/microsoft/TypeScript/blob/a9efe2be6040e581100231891dea78a7fbc4f46c/src/services/jsDoc.ts#L165

https://github.com/microsoft/TypeScript/blob/a9efe2be6040e581100231891dea78a7fbc4f46c/src/services/jsDoc.ts#L213-L214

Reply:
@DanielRosenwasser Do I need to add additional changes?